modules/@apostrophecms/doc-type/index.js

```diff
@@ -502,6 +503,23 @@ module.exports = {
         }
         return self.apos.migration.addSortify(self.__meta.name, { type: self.name }, field);
       },
+      // Add the "cacheInvalidatedAt" field to the documents that do not have it yet,
+      // and set it to equal doc.updatedAt.
+      addCacheFieldMigration() {
+        self.apos.migration.add(`add-cache-invalidated-at-field-for-${self.__meta.name}`, () => {
```

Comment:
That's a lot of migrations and the behavior doesn't differ from one doc type to another. So write just one migration in `@apostrophecms/doc` which operates on all docs.